lfshttp/certs_darwin.go

```diff
@@ -68,5 +70,60 @@ func appendRootCAsFromKeychain(pool *x509.CertPool, name, keychain string) *x509
 		tracerx.Printf("Error reading keychain %q: %v", keychain, err)
 		return pool
 	}
-	return appendCertsFromPEMData(pool, data)
+
+	// /usr/bin/security find-certificate does not support exact matching, so we have to do
+	// some manual filtering.
+	return appendMatchingCertsByCN(pool, data, hostname)
+}
+
+// appendRootCAsFromKeychain adds certificates from the specified keychain to the certificate pool
+// for the given hostname. It filters out CA certificates that have the hostname as a substring
+// in their CN but aren't an exact match for the hostname. This prevents issues where searching
+// for "hostname" returns certificates like "hostname Some Authority" that shouldn't be used
+// for TLS verification of the actual hostname.
+func appendMatchingCertsByCN(pool *x509.CertPool, data []byte, hostname string) *x509.CertPool {
+	if len(data) == 0 {
+		return pool
+	}
+
+	// Parse PEM data into individual certificates
+	pemBlocks := []byte(data)
+	var validCerts [][]byte
+
+	for len(pemBlocks) > 0 {
+		var block *pem.Block
+		block, pemBlocks = pem.Decode(pemBlocks)
+		if block == nil {
+			break
+		}
+		if block.Type != "CERTIFICATE" {
+			continue
+		}
+
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			continue
+		}
+
+		// Skip CA certificates where the hostname is a substring of the CN
+		// but not an exact match (like "example.com JSS Built-in Certificate Authority")
+		if cert.IsCA &&
+			strings.Contains(cert.Subject.CommonName, hostname) &&
+			!strings.EqualFold(cert.Subject.CommonName, hostname) {
```

Comment:
Hmm, this is problematic if you happen to have a TLS cert that requires this bundle:

1. `example.org`
2. `example.org Intermediate CA`
3. `example.org Root CA`

It's possible all three certs are needed here? I guess this is really a problem with `/usr/bin/security find-certificate` not doing exact matching, or these intermediate certificates having a misleading CN?